src/print.S

```diff
@@ -62,57 +270,506 @@ print_int:
     imul %ebx, %r10d
     subq %r10, %r9
 
+    incl %r12d
     cmpq $0, %r8
-    ja .Lprint_int_loop
-
+    ja .Lprint_number_loop
+    cmpl %r13d, %r14d
+    jge .Lprint_number_return
+    movl %r14d, %edi
+    subl %r13d, %edi
+    callq append_zeros
+.Lprint_number_return:
+    movl $0x0A, %eax
+    movl %eax, -4(%rbp)
+    leaq -4(%rbp), %rdi
+    callq print_char
     nop
-    popq %r10
-    popq %r9
-    popq %r8
+    popq %r13
+    popq %r12
+    popq %rbx
     movq %rbp, %rsp
     popq %rbp
     retq
 
 /* ======================================================
- * print a single character using the write syscall
+ * Split the IEEE754 floating-point number into it's sub parts:
+ * Sign bit, exponent and mantissa.
+ *
  * parameter:
- *  rdi => pointer to character
+ *  rdi => pointer to structure:
+ *      +0 => sign bit
+ *      +4 => exponent
+ *      +8 => mantissa
+ *
+ * parameter:
+ *  xmm0 => input floating-point number
  */
-print_char:
-    movq %rdi, %rsi /* msg argument of write syscall */
-    movq $0x01, %rax /* write syscall */
-    movq $0x01, %rdi /* write to stdout */
-    movq $0x01, %rdx /* length to write (1) */
-    syscall
+split_into_parts:
+    pushq %rbp
+    movq %rsp, %rbp
+    vmovss %xmm0, -8(%rbp)
+    leaq -8(%rbp), %rax
+    movl (%rax), %eax
+    /* 1. sign bit */
+    sarl $31, %eax
+    movzbl %al, %eax
+    movl %eax, (%rdi)
+    /* 2. exponent */
+    leaq -8(%rbp), %rax
+    movl (%rax), %eax
```

Comment:
Again, this could be:

```asm
movl -8(%rbp), %eax
```

But, alas, even better to have previously saved %eax in %edx or similar, in order not to read from memory again.